packages/firestore/test/unit/remote/remote_event.test.ts

```diff
@@ -345,186 +446,178 @@ describe('RemoteEvent', () => {
+        aggregator.addDocument(1, updatedDoc2);
+        aggregator.addDocument(1, doc3);
+
+        event = aggregator.createRemoteEvent(version(3));
+
+        expectEqual(event.snapshotVersion, version(3));
+        expect(event.documentUpdates.size).to.equal(3);
+        // Doc 1 is replaced
+        const olddoc = event.documentUpdates.get(doc1.key);
+        expectEqual(olddoc, deletedDoc1);
+        // Doc 3 is new
+        expectEqual(event.documentUpdates.get(doc3.key), doc3);
+
+        // Target is unchanged
+        expect(size(event.targetChanges)).to.equal(1);
+
+        const mapping1 = updateMapping(
+          version(3),
+          [doc3],
+          [updatedDoc2],
+          [deletedDoc1]
+        );
+        expectTargetChangeEquals(event.targetChanges[1], mapping1);
+      }
+    );
   });
 
   it('synthesizes deletes', () => {
-    const targets = listens(1);
-    const shouldSynthesize = new WatchTargetChange(
+    const targets = limboListens(1);
+    const limboKey = DocumentKey.fromPathString('coll/limbo');
+    const resolveLimboTarget = new WatchTargetChange(
       WatchTargetChangeState.Current,
       [1]
     );
 
-    const event = remoteEvent(1, targets, noPendingResponses, shouldSynthesize);
-
-    const synthesized = DocumentKey.fromPathString('docs/2');
-    expect(event.documentUpdates.get(synthesized)).to.be.null;
-
-    const limboTargetChange = event.targetChanges[1];
-    event.synthesizeDeleteForLimboTargetChange(limboTargetChange, synthesized);
-    const expected = deletedDoc(
-      'docs/2',
-      event.snapshotVersion.toMicroseconds()
+    withRemoteEvent(
+      1,
+      targets,
+      noPendingResponses,
+      noExistingKeys,
+      [resolveLimboTarget],
+      event => {
+        const expected = deletedDoc(
+          'coll/limbo',
+          event.snapshotVersion.toMicroseconds()
+        );
+        expectEqual(event.documentUpdates.get(limboKey), expected);
+        expect(event.resolvedLimboDocuments.has(limboKey)).to.be.true;
+      }
     );
-    expectEqual(event.documentUpdates.get(synthesized), expected);
-    expect(event.limboDocuments.has(synthesized)).to.be.true;
   });
 
   it("doesn't synthesize deletes in the wrong state", () => {
+    const targets = limboListens(1);
+    const limboKey = DocumentKey.fromPathString('coll/limbo');
     const wrongState = new WatchTargetChange(WatchTargetChangeState.NoChange, [
-      2
+      1
     ]);
-    const targets = listens(2);
-    const event = remoteEvent(1, targets, noPendingResponses, wrongState);
-
-    const notSynthesized = DocumentKey.fromPathString('docs/no1');
-    const wrongStateChange = event.targetChanges[2];
-    event.synthesizeDeleteForLimboTargetChange(
-      wrongStateChange,
-      notSynthesized
-    );
-    expect(event.documentUpdates.get(notSynthesized)).to.not.exist;
-    expect(event.limboDocuments.has(notSynthesized)).to.be.false;
-  });
 
-  it("doesn't synthesize deletes with existing document", () => {
```

Comment:
This test is not applicable anymore since we always synthesize deletes for single document lookups (provided we are marked current).